Add multi-threading variants to sdl2_mixer

db-tech opened the pull request:
I added support for multi threading (-pthread) to the sdl2_mixer port. 
For me this was kind of new but I did try to follow the prinziples of the sdl2_image adn sdl2_ttf (https://github.com/emscripten-core/emscripten/pull/22946) changes.

I also changed the "_" to "-" for consistency.

Let me know if there is something wrong or missing.

sbc100 commented:
Seems reasonable to me.  Does SDL2_mixer actually need thread support?

db-tech commented:
Should fix some dependency issues discussed here: https://github.com/emscripten-core/emscripten/issues/23093#

To summerize:
Without the changes I get errors like:

`AssertionError: attempt to lock the cache while a parent process is holding the lock (sysroot/lib/wasm32-emscripten/libSDL2.a)`

So It might be a dependency problem because mixer depends on another SDL2 version (-mt) then the other SDL2 libs and would differ from the SDL2 lib that is used when -pthread is enabled.

ololoken reviewed (commented):
lgtm
tested and fix works

sbc100 commented on tools/ports/sdl2_mixer.py:
```diff
@@ -10,8 +10,10 @@
-  'sdl2_mixer_none': {'SDL2_MIXER_FORMATS': []},
+  'sdl2_mixer-mp3': {'SDL2_MIXER_FORMATS': ["mp3"]},
+  'sdl2_mixer-none': {'SDL2_MIXER_FORMATS': []},
+  'sdl2_mixer-mp3-mt': {'SDL2_MIXER_FORMATS': ["mp3"], 'PTHREADS': 1},
```
Can you use single quote here and above.

sbc100 commented:
It would be good to a test here that depends on this new variant.   What is the set of command line flags you used to cause the `AssertionError: attempt to lock the cache while a parent process is holding the lock (sysroot/lib/wasm32-emscripten/libSDL2.a)` failure?

sbc100 commented:
Also can you rebase or merge so you are up-to-date with main?

oleg-derevenetz commented:
Hi @sbc100

> What is the set of command line flags you used to cause the AssertionError: attempt to lock the cache while a parent process is holding the lock (sysroot/lib/wasm32-emscripten/libSDL2.a) failure?

Here is a reproducible example of the build failure when building the project that depends on sdl2-mixer with `-pthread`:

https://github.com/oleg-derevenetz/fheroes2/actions/runs/12609145320/job/35142382904#step:5:40

and here is the corresponding test PR:

https://github.com/oleg-derevenetz/fheroes2/pull/218

If there are any issues with maintaining this PR, I can create my own (based on this one with credits to @db-tech) to speed up the process, because I'm interested in solving this build issue, since I would like to build my project for WASM with threads... eventually.

db-tech commented:
Hi, sorry for not answering.
I am currently on vacation and usually off the grid.

Tomorrow, I might have time to address the comments. 

Otherwise, I am back next week.

db-tech commented:
As @oleg-derevenetz said "-pthread" is the option which results in issues.
I changed the double quotes to single ones and, fetched upstream and rebased, I hope that was correct.
Otherwise, let me know.
